Book creation

bulatovicb opened the pull request:
https://matijabojanic.atlassian.net/browse/OL3-18

bulatovicb committed:
Book creation

https://matijabojanic.atlassian.net/browse/OL3-18

Ana0312 commented on app/Http/Controllers/BookController.php:
```diff
@@ -2,9 +2,103 @@
+            'number_of_pages' => 'required',
+            'number_of_copies_available' => 'required',
+            'isbn' => 'required',
+            'language' => 'nullable',
```
Primijetila sam da u migraciji polja language, script, binding, dimensions nisu oznacena kao nullable pa dodaj migraciju koja mijenja ta polja da budu nullable. Treba da bude konzistentno sa ovim validatorom kako si napravila

Ana0312 commented on app/Http/Controllers/BookController.php:
```diff
@@ -2,9 +2,103 @@
+
+                $path = $image->store('book_images', 'public');
+
+
```
Imas na par mjesta ove prazne linije, vjerovatno greskom, pa samo ispravi to, nema potrebe na pocetku funcije da bude prazno na pocetku f-je, na par mjesta ces vidjeti da imas prazne linije.

Ana0312 commented on app/Http/Controllers/BookController.php:
```diff
@@ -2,9 +2,103 @@
+            'images' => 'nullable|array',
+            'images.*' => 'file|image|max:5120',
+            'image_types' => 'nullable|array',
+            'image_types.*' => ['required', Rule::in(['front_cover', 'back_cover', 'artwork'])],
```
Mozes napraviti konstantu za ovaj niz u Image modelu jer na vise mjesta pozivas vjerovatno

Ana0312 commented on app/Http/Controllers/BookController.php:
```diff
@@ -2,9 +2,103 @@
+        $book->publishers()->attach($request->publishers);
+
+        $book->load(['images', 'authors']);
+
```
Mozes ukloniti jednu praznu liniju

bulatovicb committed:
Add Const for images

Add new migration to make fields nullable

https://matijabojanic.atlassian.net/browse/OL3-18

Ana0312 reviewed (approved):
Code reviewed, looks good

bulatovicb committed:
Merge branch 'book-management' into task/OL3-3/OL3-18/Book-creation
